### ISSUE TITLE
Code that uses "crypto" module fails on server due to resolving to "crypto-browserify".

### ISSUE THREAD

**Issue by thesunny** [thereactivestack/meteor-webpack#54]
"meteor-webpack" has a dependency on "crypto-browserify" and that module is used to satisfy any imports or requires, even on the server.

Whenever you do

```javascript
var crypto = require("crypto");
```

It grabs the "crypto-browserify" version even on the server. On the server, it should be grabbing "crypto" from node.

My use case: I'm trying to use the "cloudinary" module on the server. It makes a call to `crypto.randomBytes` and then throws this error:

```
Exception while invoking method 'uploadBytesToCloudinary'
Error: secure random number generation not supported by this browser
use chrome, FireFox or Internet Explorer 11
```

The stack trace shows it is coming from "crypto-browserify"

```
at module.exports (webpack:///(webpack)/~/node-libs-browser/~/crypto-browserify/rng.js:22:1)
```

I tried to work around it through webpack configuration but wasn't successful at it.

Any suggested work arounds or a fix would be greatly appreciated (spent several hours on this).

**Comment by thesunny** [comment]
I just wanted to comment that I did find a workaround that I recommend incorporating in to meteor-webpack itself.

Basically for "webpack.conf.js" on the server, you have to specify that crypto should be an external that is provided through commonjs. I basically tried everything until something worked so, even after reading the documentaiton, it wasn't clear that this was the solution.

Here is the supporting documentation:

https://webpack.github.io/docs/configuration.html#externals

Make sure to see the section on "commonjs"

Then for webpack.conf.js on the server, add this configuration:

```javascript
module.exports = {
  externals: {
    // other externals configuration here...
    'crypto': 'commonjs crypto'
  },
  // other configuration here...
}
```

**Comment by eXon** [comment]
If you want to run a lib on the server, you should use `meteorhacks:npm` package instead.

**Comment by thesunny** [comment]
Thanks for the suggestion. That's ultimately what I ended up doing.

It feels a little dirty though as, I'm assuming, that any shared libraries between using Npm.require and using import on the server will be double loaded.

**Comment by eXon** [comment]
For now the server-code is bundled completely but I'm looking for a way to set the target to node.js. It could be easier.

**Comment by eXon** [comment]
This will be fixed on 1.0. Release should be march 1st.
